packages/checkout-extension/src/ExtensionService.ts

```diff
@@ -58,9 +58,18 @@ export class ExtensionService {
             return;
         }
 
-        await this.checkoutService.renderExtension(container, region);
-
-        this.registerHandlers(extension);
+        try {
+            await this.checkoutService.renderExtension(container, region);
+
+            this.registerHandlers(extension);
+        } catch (error: unknown) {
+            if (error instanceof Error) {
+                // eslint-disable-next-line no-console
+                console.error(
```

Comment:
Shall we use `ErrorLogger` here?

Reply:
I understand `ErrorLogger` ends up at Sentry.
An extension iframe's detachment error may happen regularly, shall we just keep it in the console?
I see we have `ConsoleErrorLogger`, I can replace here with that.

Reply:
Thanks for the feedback, I have replaced it.
This is an updated testing screenshot.
<img width="1564" alt="image" src="https://github.com/user-attachments/assets/1b466e70-a71a-40af-9cbe-338a41eb816b">

Reply:
@bc-peng Have you figured out why `Shipping` component is momentarily rendered at all? I'd expect that after customer/shipping/billing details are provided, when you refresh the page, we should only render the `Payment` component, not the `Shipping` component I suspect it is probably because for some reason `isComplete` is `false` here: https://github.com/bigcommerce/checkout-js/blob/d6d1218e0628e462d79bfc36fdd278c7936663ea/packages/core/src/app/checkout/getCheckoutStepStatuses.ts#L181

But it shouldn't be because shipping details are already provided.

Having said that, we need to have what you've done so far as a safety net - just in case the extensions are detached while loading. However, it seems something is wrong with the state machine that also needs fixing.

Reply:
Good catch @davidchin I think might be good to check and test as part of this PR itself. It could be because of checking for additional conditions like fastlane etc. Or could it be that since we changed for extension to start loading while we fetch other shipping data if customer email already complete?

Reply:
For the record, we have implemented this change to prevent the shipping form from rendering after a page refresh.

https://github.com/bigcommerce/checkout-js/pull/2111